app/lib/constituency_api.rb

```diff
@@ -0,0 +1,51 @@
+module ConstituencyApi
+  ConstituencyApiError = Class.new(RuntimeError)
+  
+  class Constituency
+    attr_accessor :name
+    
+    def initialize(name)
+      self.name = name
+    end
+    
+    def ==(another_constituency)
+      self.name == another_constituency.name
+    end
+  end
+
+  class Client
+    include Faraday
+    URL = 'http://data.parliament.uk/membersdataplatform/services/mnis/Constituencies'
+    TIMEOUT = 10
+    
+    def self.constituencies(postcode)
+      response = call_api(postcode)
+      parse_constituencies(response)
+    end
+    
+    private
+    
+    def self.parse_constituencies(response)
+      return [] unless response["Constituencies"]
+      constituencies = response["Constituencies"]["Constituency"]
+      Array.wrap(constituencies).map { |c| Constituency.new(c["Name"]) }
+    end
+    
+    def self.call_api(postcode)
+      response = Faraday.new(URL).get("#{postcode_param(postcode)}/") do |req|
+        req.options[:timeout] = TIMEOUT
+        req.options[:open_timeout] = TIMEOUT
+      end
+      raise ConstituencyApiError.new("Unexpected response") unless response.status == 200
```

Comment:
We could include some more information on this error message. When the API is changed (or there is a special case we haven't taken into account), our wrapper breaks and we would then have meaningful log messages to help us fix the issue. 

```ConstituencyApiError.new("Unexpected response from API: status #{response.status} body #{response.body}")```? And maybe even include the full request url in the message.